### PR TITLE
Correctly pass `newline` parameter to built-in `open` function

### LINKED ISSUE #477: Output using `csv` module is inconsistent with traditional `open` behaviour
#### Problem description

I'm using python 3.7 and `smart_open==1.11.1` under Windows 10. There is [this issue with python, csv, and Windows](https://stackoverflow.com/questions/3191528/csv-in-python-adding-an-extra-carriage-return-on-windows) which your would normally solve by the mentioned solution.

However, if I replace `open` with `smart_open.open` there seems to be a different behaviour. `smart_open` still adds new lines to the csv, as described in the following test.
 
#### Steps/code to reproduce the problem

```
platform win32 -- Python 3.7.5, pytest-5.4.1, py-1.8.1, pluggy-0.13.1 -- 
...
cachedir: .pytest_cache
...
plugins: cov-2.8.1, mypy-0.6.0
collecting ... collected 1 item

tests/test_io.py::test_compatibility_with_csv_under_windows FAILED       [100%]
tests\test_io.py:53 (test_compatibility_with_csv_under_windows)
c1,c2

a1,b1

a2,b2

 != c1,c2
a1,b1
a2,b2


def test_compatibility_with_csv_under_windows():
        rows = [{'c1': 'a1', 'c2': 'b1'}, {'c1': 'a2', 'c2': 'b2'}]
    
        expected = 'c1,c2\na1,b1\na2,b2\n'
    
        with open('tests/test.csv', 'w+', newline='\n') as f:
            out = csv.DictWriter(f, fieldnames=['c1', 'c2'])
            out.writeheader()
            out.writerows(rows)
    
        with smart_open.open('tests/test_smart_open.csv', 'w+', newline='\n') as f:
            out = csv.DictWriter(f, fieldnames=['c1', 'c2'])
            out.writeheader()
            out.writerows(rows)
    
    
        with open('tests/test.csv', 'r') as f:
            content = f.read()
            assert content == expected
    
        with open('tests/test_smart_open.csv', 'r') as f:
            content = f.read()
>           assert content == expected
E           AssertionError: assert 'c1,c2\n\na1,b1\n\na2,b2\n\n' == 'c1,c2\na1,b1\na2,b2\n'
```

#### Versions

Please provide the output of:

```python
Windows-10-10.0.17134-SP0
Python 3.7.5 (tags/v3.7.5:5c02a39a0b, Oct 15 2019, 00:11:34) [MSC v.1916 64 bit (AMD64)]
smart_open 1.11.1
```

#### Checklist

Before you create the issue, please make sure you have:

- [ ] Described the problem clearly (hope so ❤️ )
- [x] Provided a minimal reproducible example, including any required data
- [x] Provided the version numbers of the relevant software

### PR DESCRIPTION
**Pull request by burkovae**
#### Motivation

It appears, that `newline` parameter is missing when smart_open uses the buildin `open` method.
This one is essential, e.g.  [when writing csv files under windows](https://stackoverflow.com/questions/3191528/csv-in-python-adding-an-extra-carriage-return-on-windows)

- Fixes #477

#### Tests

See test case.

* [ ] Please @maintainers, run the complete test suite, because (i) I lack the infrastructre for testing google and aws stuff (tests are not passing from master branch) and (ii) I'm not able to test against Macs and Unix systems.
* [ ] Please review the parameter position. I put it to the end, so that nothing will break in the first palce. But I may be sensible [to put it into the same place as open would expect.](https://docs.python.org/3/library/functions.html#open)
* [ ] I noticed unused parameter `kw` and removed it as well, no harm done ❤️ 

#### Checklist

Before you create the PR, please make sure you have:

- [ ] Picked a concise, informative and complete title (hopefully 🙉 )
- [x] Clearly explained the motivation behind the PR
- [x] Linked to any existing issues that your PR will be solving
- [x] Included tests for any new functionality
- [ ] Checked that all unit tests pass (need support from maintainer here 🥇  )

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -68,7 +68,7 @@ def read(fname):
 
     packages=find_packages(),
     package_data={
-        "smart_open.tests": ["test_data/*gz"],
```
Comment (burkovae) [inline setup.py:71]: Wasn't me ;-( - where does it come from ?

Comment (mpenkov) [reply setup.py]: No big deal. You branched your work from the master branch, but you're trying to merge it to the develop branch. This is why your PR includes work from other people. I rebased your PR off develop.

In the future, please branch off develop.

**Comment by mpenkov** [comment]
I've pushed some changes to polish up your PR. Can you please run the tests on your end and confirm that they pass?

In the future, we'll really need to run tests on Windows as part of our CI. https://github.com/RaRe-Technologies/smart_open/issues/161

**Comment by burkovae** [comment]
Thank you for the edits. I now remember, why [temprorary files](https://github.com/bravoserver/bravo/issues/111) are [quite nasty under Windows 🔥](https://docs.python.org/3/library/tempfile.html#tempfile.NamedTemporaryFile)

> Whether the name can be used to open the file a second time, while the named temporary file is still open, varies across platforms (it can be so used on Unix; it cannot on Windows NT or later)

This is the reason, why the test fails atm.

What do you think ?

**Comment by mpenkov** [comment]
Can you please tweak the test so that it passes? I can't reproduce the problem without Windows, unfortunately. I'd do something like this:

```python
import contextlib

@contextlib.contextmanager
def win_compatible_tempfile(mode):
    with tempfile.NamedTemporaryFile(mode, delete=False) as f:
        try:
            yield f
        finally:
            f.close()
            os.unlink(f.name)
```

based on this comment https://github.com/bravoserver/bravo/issues/111#issuecomment-826990
The reason I use tempfile is because I want to avoid writing to the present working directory. It's poor practice.

**Comment by mpenkov** [comment]
Please pull from develop and look for a function called named_temporary_file. Use it to work around the problem.

**Comment by burkovae** [comment]
Thank you for your suggestions and improvements. The test is green (at least locally.). Sorry, it took so long to respond.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,7 +1,11 @@
 # Unreleased
 
+<<<<<<< HEAD
```
Comment (mpenkov) [inline CHANGELOG.md:3]: This looks like a merge conflict

Comment (burkovae) [reply CHANGELOG.md]: Oi! Thanks, strangely `git status` didn't told me.

**Comment by mpenkov** [comment]
Finally got around to merging this. Good work @burkovae , and congrats on your first PR to smart_open! :1st_place_medal:

**Pull request merged**